src/TestEnvironment.Docker/DockerApi/Internal/Services/DockerImagesService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Docker.DotNet;
+using Docker.DotNet.Models;
+using TestEnvironment.Docker.DockerApi.Abstractions.Models;
+using TestEnvironment.Docker.DockerApi.Abstractions.Services;
+
+namespace TestEnvironment.Docker.DockerApi.Internal.Services
+{
+    public class DockerImagesService : IDockerImagesService
+    {
+        private readonly DockerClient _client;
+
+        public DockerImagesService(DockerClient client)
+        {
+            _client = client;
+        }
+
+        public async Task BuildImage(ImageFromDockerfileConfiguration configuration, string tempFileName, CancellationToken cancellationToken)
```

Comment:
I think it would be better to move logic responsible for archiving context inside this method.